src/cascadia/TerminalConnection/AzureConnection.cpp

```diff
@@ -0,0 +1,885 @@
+                        pplx::task<websocket_incoming_message> msgT;
+                        try
+                        {
+                            msgT = _cloudShellSocket.receive();
+                            msgT.wait();
+                        }
+                        catch (...)
+                        {
+                            // Websocket has been closed
+                            //_termConnected = false;
+                            if (!_closing.load())
+                            {
+                                _disconnectHandlers();
+                                return S_FALSE;
+                            }
+                            break;
+                        }
+                        auto msg = msgT.get();
+                        auto msgStringTask = msg.extract_string();
+                        auto msgString = msgStringTask.get();
+
+                        // Convert to hstring
+                        const auto hstr = winrt::to_hstring(msgString);
+
+                        // Pass the output to our registered event handlers
+                        _outputHandlers(hstr);
+                    }
+                    return S_OK;
+                }
+                case State::noConnect:
+                {
+                    _outputHandlers(winrt::to_hstring(internetOrServerIssue));
+                    return E_FAIL;
+                }
+                }
+            }
+            catch (...)
+            {
+                _state = State::noConnect;
+            }
+        }
+    }
+
+    // Method description:
+    // - helper function to get the stored credentials (if any) and let the user choose what to do next
+    // Return value:
+    // - S_FALSE if there are no stored credentials
+    // - S_OK if the user opts to login with a stored set of credentials or login with a different account
+    // - E_FAIL if the user closes the tab
+    HRESULT AzureConnection::_accessHelper()
+    {
+        auto vault = PasswordVault();
+        winrt::Windows::Foundation::Collections::IVectorView<PasswordCredential> credList;
+        // FindAllByResource throws an exception if there are no credentials stored under the given resource so we wrap it in a try-catch block
+        try
+        {
+            credList = vault.FindAllByResource(resource);
+        }
+        catch (...)
+        {
+            // No credentials are stored, so start the device flow
+            _state = State::deviceFlow;
+            return S_FALSE;
+        }
+        _maxStored = credList.Size();
+        // Display the user's saved connection settings
+        for (int i = 0; i < _maxStored; i++)
+        {
+            auto entry = credList.GetAt(i);
+            auto nameJson = json::value::parse(entry.UserName().c_str());
+            _outputHandlers(_strFormatHelper(ithTenant, i, nameJson.at(L"displayName").as_string().c_str(), nameJson.at(L"tenantID").as_string().c_str()));

```

Comment:
I might be insane but where is ithTenant declared?

Reply:
In AzureConnectionStrings.h, ithTenant is the string format